Send clientInfo during the initialize request

nemethf opened the pull request:
I haven't tried out native compilation, but my guess is this PR will work there as well.

----

clientInfo arrived in LSP 3.15.0.  It is generally useful for various
debugging tasks.

* eglot.el (eglot--version): New defconst.
(eglot--connect): Send clientInfo using eglot--version.

skangas commented:
Thanks. Should this have a unit test?

nemethf commented:
> Thanks. Should this have a unit test?

I force pushed a new version with a test.  I wrote the test in a cargo cult style, but it seems to work.  By the way, am I reading it correctly that it's not possible to run just a single test with the current Makefile?  That would be useful during development.  Or is there another way to do it?

skangas commented:
Felicián Németh ***@***.***> writes:

> I force pushed a new version with a test. I wrote the test in a cargo cult
> style, but it seems to work.
>
Thanks!

> By the way, am I reading it correctly that it's not possible to run just a
> single test with the current Makefile? That would be useful during
> development. Or is there another way to do it?
>
Yes, that would be useful. Check out the ERT manual (see the section on
test selectors) and/or the file test/Makefile.in in the Emacs git repo for
how to do that. Perhaps there is nothing more we need to do, or we would
need to add some more stuff, I can't remember.

If there is more to do, patches to add such support would definitely be
welcome. Otherwise, perhaps we could get away with a documentation patch to
the "Commentary" section of eglot-tests.el. Thanks in advance.

> Message ID: ***@***.***>
>

nemethf commented:
@skangas, I addressed all the issues you raised in your review.  Do you see any remaining problem with this PR?  People seem to miss this feature:
https://github.com/rust-lang/rust-analyzer/pull/14141#issuecomment-1429477377

Thanks.

joaotavora commented:
I think this is generally OK, although I'm not familiar with this auto-versioning technique in `eglot--version`.  Please submit this as a patch to the bug tracker and put me in CC.  Thanks.

nemethf commented:
Discussion continued in Emacs bug#62198.  I think commit https://github.com/emacs-mirror/emacs/commit/c5f369bacc414fce6a1ed17733188f4c1cdfe317 closes this issue (#818).
